lib/maxwell/middleware/header.ex

```diff
@@ -20,20 +20,17 @@ defmodule Maxwell.Middleware.Headers do
   alias Maxwell.Conn
 
   def init(headers) do
-    check_headers(headers)
-    Conn.new()
-    |> Conn.put_req_headers(headers)
-    |> Map.get(:req_headers)
+    check_headers!(headers)
+    headers
```

Comment:
`put_req_header` make header be downcase.
middleware's operation alway after user's operation, 
if we do this, we must make Header middleware alway before HeaderCase middleware.

User alway knows about what's put in header, so I don't check header in middleware.